File: php/features.php
<?php
declare(strict_types=1);

// IamSmart SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class IamSmartFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IamSmartBaseFeature();
            case "test":
                return new IamSmartTestFeature();
            default:
                return new IamSmartBaseFeature();
        }
    }
}

File: py/features.py
# IamSmart SDK feature factory

from feature.base_feature import IamSmartBaseFeature
from feature.test_feature import IamSmartTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IamSmartBaseFeature(),
        "test": lambda: IamSmartTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
